lua/mini/misc.lua

```diff
@@ -409,6 +413,49 @@ MiniMisc.zoom = function(buf_id, config)
   end
 end
 
+---
+MiniMisc.setup_goto_last_cursor = function()
+  if vim.fn.has("nvim-0.7.0") then
+    local group_name = "MiniMiscGotoLastCursor"
+
+    vim.api.nvim_create_augroup(group_name, { clear = true })
+    vim.api.nvim_create_autocmd("BufWinEnter", { callback = MiniMisc.goto_last_cursor, group = group_name })
+    vim.api.nvim_create_autocmd("FileType", { callback = MiniMisc.goto_last_cursor, group = group_name })
+  else
+    vim.api.nvim_command([[augroup MiniMiscGotoLastCursor]])
+    vim.api.nvim_command([[  autocmd!]])
+    vim.api.nvim_command([[  autocmd BufWinEnter * lua require('mini.misc').goto_last_cursor()]])
+    vim.api.nvim_command([[  autocmd FileType * lua require('mini.misc').goto_last_cursor()]])
+    vim.api.nvim_command([[augroup end]])
```

Comment:
You can do all this in a single `vim.api.nvim_exec([[...]], false)` call.

Is there a reason to use `BufWinEnter` event? It means cursor will be tried to position every time buffer is displayed in a window.

Is there a reason for not following an example from `:h last-position-jump`?

Reply:
Done. As you already use nvim_create_autocmd in setup_auto_root I used that

Reply:
> Done. As you already use nvim_create_autocmd in setup_auto_root I used that

`nvim_create_autocmd()` in `setup_auto_root()` was used because it already relies on Neovim>=0.8. This new function doesn't need to rely on that version and can support Neovim>=0.6. Let's do a more reverse compatible version for now.

Reply:
> I've tried to do a one line autcomd but couldn't get it working. This worked just fine. As it is really new functionality why not used the latest API if misc already uses that=

Only `setup_auto_root()` and `find_root()` depend on having `vim.fs` (the same as depending on Neovim>=0.8). All other 'mini.misc' functions support Neovim>=0.6.